samples/apps/copilot-chat-app/webapp/src/components/chat/ChatInput.tsx

```diff
@@ -160,16 +170,41 @@ export const ChatInput: React.FC<ChatInputProps> = (props) => {
         }
     };
 
+    const [isDraggingOver, setIsDraggingOver] = React.useState(false);
+
+    const handleDragEnter = (e: React.DragEvent<HTMLTextAreaElement>) => {
+        e.preventDefault();
+        setIsDraggingOver(true);
+    };
+
+    const handleDragLeave = (e: React.DragEvent<HTMLTextAreaElement>) => {
+        e.preventDefault();
+        setIsDraggingOver(false);
+    };
+
+    const handleDrop = async (e: React.DragEvent<HTMLTextAreaElement>) => {
+        handleDragLeave(e);
+        await importDocument(e.dataTransfer?.files[0]);
+    };
+
     return (
         <div className={classes.root}>
             <div className={classes.typingIndicator}>{isTyping ? <TypingIndicatorRenderer /> : null}</div>
             <div className={classes.content}>
                 <Textarea
                     id="chat-input"
                     resize="vertical"
-                    textarea={{ className: classes.textarea }}
+                    textarea={{
+                        className: isDraggingOver
+                            ? mergeClasses(classes.dragAndDrop, classes.textarea)
+                            : classes.textarea,
+                    }}
                     className={classes.input}
-                    value={value}
+                    value={isDraggingOver ? 'Drop your files here' : value}
+                    onDragEnter={handleDragEnter}
+                    onDragOver={handleDragEnter}
+                    onDragLeave={handleDragLeave}
```

Comment:
nit: `onDragEnter` and `onDragOver` should be properties on the ChatRoom component so the user knows where to drop the file upon entering the view

Reply:
good idea, moved